doc/man3/SSL_set_async_callback.pod

```diff
@@ -0,0 +1,101 @@
+=pod
+
+=head1 NAME
+
+SSL_CTX_set_async_callback,
+SSL_CTX_set_async_callback_arg,
+SSL_set_async_callback,
+SSL_set_async_callback_arg,
+SSL_get_async_status,
+SSL_clear_async_status,
+SSL_async_callback_fn
+- manage asynchronous operations
+
+=head1 SYNOPSIS
+
+=for comment multiple includes
+
+ #include <openssl/ssl.h>
+
+ typedef int (*SSL_async_callback_fn)(SSL *s, void *arg);
+ int SSL_CTX_set_async_callback(SSL_CTX *ctx, SSL_async_callback_fn callback);
+ int SSL_CTX_set_async_callback_arg(SSL_CTX *ctx, void *arg);
+ int SSL_set_async_callback(SSL *s, SSL_async_callback_fn callback);
+ int SSL_set_async_callback_arg(SSL *s, void *arg);
+ int SSL_get_async_status(SSL *s, int *status);
+ int SSL_clear_async_status(SSL *s);
+
+=head1 DESCRIPTION
+
+SSL_CTX_set_async_callback() sets asynchronous callback function, and all SSL
+sessions generated based on this SSL_CTX can get this callback, which will be
+automatically called when engine completed the cryptography operation to notify
+the application to resume the paused work flow.
+
+SSL_CTX_set_async_callback_arg() sets the callback argument.
+
+SSL_set_async_callback() allows a SSL application to set a callback in an
+asynchronous SSL session, so that when the engine completed a cryptography
+operation, the callback can be called to notify the application to resume the
+paused work flow.
+
+SSL_set_async_callback_arg() helps to set an argument for the SSL session when
+the above callback is called.
+
+SSL_get_async_status() returns the engine status. This function facilitates the
+communication from the engine to the application. During an SSL session,
+cryptographic operation would be dispatched to the engine. The engine status is
+very useful for an application to know if the engine has successfully dispatched
+it to hardware. If the engine does not support this additional callback method,
+"ASYNC_STATUS_UNSUPPORTED" will be returned.
+
+SSL_clear_async_status() helps SSL application to clear the engine status if the
+action has been taken already to avoid duplication handing.
+
+An example of the above functions would be the following.
+1. Application sets the async callback and callback data on an SSL connection
+by calling SSL_set_async_callback().
+2. Application makes an asynchronous SSL call
+3. OpenSSL submits the asynchronous request within the OpenSSL engine. If a
+retry occurs at this point then the status within the ASYNC_WAIT_CTX would be
+set and the async callback function would be called (goto Step 7).
+4. The OpenSSL engine pauses current job and return, so the application can
```

Comment:
"pauses current job and return, so the application" -> "pauses the current job and returns, so that the application"